src/main/java/thewaypointers/trafficsimulator/gui/MainFrameEventHandle.java

```diff
@@ -1,31 +1,47 @@
 package thewaypointers.trafficsimulator.gui;
 
+import sun.applet.Main;
 import thewaypointers.trafficsimulator.JunctionLocationTestStarter;
 
+import java.awt.*;
 import java.awt.event.ComponentEvent;
 import java.awt.event.ComponentListener;
 
 public class MainFrameEventHandle implements ComponentListener {
+
+    private MainFrame processComponent(Component c){
+        if (!c.getClass().equals(MainFrame.class)){
+            throw new AssertionError(String.format(
+                    "%s being used for %s instead of %s",
+                    getClass().getName(),
+                    c.getClass().getName(),
+                    MainFrame.class.getName()
+            ));
+        }
+        return (MainFrame) c;
+    }
+
     @Override
     public void componentResized(ComponentEvent e) {
-        JunctionLocationTestStarter.mainFrame.setLayout(null);
-        MainFrame.mapContainerPanel.setBounds(0,0, JunctionLocationTestStarter.mainFrame.getWidth()-200, JunctionLocationTestStarter.mainFrame.getHeight()-60);
-        MainFrame.controlPanel.setBounds(JunctionLocationTestStarter.mainFrame.getWidth()-200,0,200, JunctionLocationTestStarter.mainFrame.getHeight());
-        MainFrame.timeLabelPanel.setBounds(0,JunctionLocationTestStarter.mainFrame.getHeight()-60,JunctionLocationTestStarter.mainFrame.getWidth()-200,60);
```

Comment:
This is very bad. It uses static variables of a particular starter class deep in the GUI code. The starter classes are just for starting with particular parameters, they should be on top and nothing should reference it. When you update or change or remove them, you shouldn't suddenly have to fix something in the GUI, this is unnecesary work for others. A much easier way to get the `MainFrame` is from `ComponentEvent`, because it has the reference to the object that the `ComponentListener` is attached to (the `MainFrame` in this case).